src/zInit.cpp

```diff
@@ -23,7 +23,7 @@ class LateInitializer {
             dlopen(dl_info.dli_fname, RTLD_LAZY | RTLD_NODELETE);
         }
 
-        if (!checkJvmLoaded()) {
+        if (!checkJvmLoaded() && OS::checkPreloaded()) {
```

Comment:
A minor thing, but I'd propose to move this check _after_ `ASPROF_COMMAND` check, since using async-profiler through native API will be a common thing, while `LD_PRELOAD` is used rarely and mostly for ad-hoc profiling only.